core/undo_redo.cpp

```diff
@@ -317,6 +317,7 @@ void UndoRedo::undo() {
 	if (current_action < 0)
 		return; //nothing to redo
 	_process_operation_list(actions[current_action].undo_ops.front());
+	ERR_FAIL_COND(current_action == -1);
```

Comment:
I still don't get the point here, `_process_operation_list` does not change `current_action` AFAIK, and the previous `current_action < 0` return already ensures that you can't reach this line with `current_action == -1`.

Reply:
Nevermind, just read the above discussion :)

Reply:
It's still too hacky though, you're applying band-aid to prevent a crash, but what needs to be fixed is that the `clear_history` logic should not be able to lead to `current_action == -1`. If it can, as @groud mentions, it means that this `ERR_FAIL_COND` will spam the logs with error messages, hinting at an unfixed bug. So best fix the bug in the first place.

Reply:
@akien-mga I agree it's better to track down the bug causing the bug. I suppose it's related to unexpected object release and it's not very easy to reproduce. I'll try to reproduce it. I'm closing this PR.